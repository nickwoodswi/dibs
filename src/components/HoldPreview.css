.multi-hold-preview{
    display: flex;
    flex-wrap: wrap;
    width: 100%;
    justify-content: space-between;
    font-size: 16px;
    padding: 15px;
    background-color: black;
}