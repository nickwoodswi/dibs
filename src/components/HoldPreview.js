import React, { Component } from 'react'
import HoldPreviewComp from './HoldPreviewComp'
import './HoldPreview.css'

class HoldPreview extends Component {
    static defaultProps = {
        appState: {
            holds: []
        }
    }
    render() {
        const holds = this.props.appState.holds
        return(
            <>
            <div className="multi-hold-preview">
                {holds.map((hold, idx) => {
                    return(
                    <HoldPreviewComp
                        key={idx}
                        startMonthPrev={hold.holdmonth_start}
                        startDayPrev={hold.holdday_start}
                        startYearPrev={hold.holdyear_start}
                        endMonthPrev={hold.holdmonth_end}
                        endDayPrev={hold.holdday_end}
                        endYearPrev={hold.holdyear_end} 
                        deleteHold = {this.props.deleteHold} />
                    )
                })}
            </div>
            </>
        )
    }
}

export default HoldPreview