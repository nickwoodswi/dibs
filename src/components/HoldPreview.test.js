import React from 'react';
import { render } from '@testing-library/react';
import HoldPreview from './HoldPreview';
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'

test('renders hold preview component', () => {
    render(<MemoryRouter><HoldPreview /></MemoryRouter>);
});

it('HoldPreview renders the UI as expected', () => {
    const tree = renderer
        .create(<MemoryRouter><HoldPreview /></MemoryRouter>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});