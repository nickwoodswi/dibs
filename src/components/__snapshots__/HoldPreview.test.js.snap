// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HoldPreview renders the UI as expected 1`] = `
<div
  className="multi-hold-preview"
/>
`;
